### PR TITLE
NO-JIRA: cleaning up aks guest annotations logic for hypershift

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aks/hypershift/heterogeneous/provision/cucushift-installer-rehearse-azure-aks-hypershift-heterogeneous-provision-chain.yaml [hunk 1]
```diff
@@ -44,5 +43,7 @@ chain:
     default: "2"
   - name: HYPERSHIFT_OPERATOR_PULL_SECRET
     default: "true"
+  - name: POD_SECURITY_ADMISSION_ANNOTATION
+    default: "false"
```
Comment (fxierh) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/aks/hypershift/heterogeneous/provision/cucushift-installer-rehearse-azure-aks-hypershift-heterogeneous-provision-chain.yaml:47]: This one does need POD_SECURITY_ADMISSION_ANNOTATION to be true imo

Comment (Patryk-Stefanski) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/aks/hypershift/heterogeneous/provision/cucushift-installer-rehearse-azure-aks-hypershift-heterogeneous-provision-chain.yaml:47]: Ill change it, I was just setting them wherever we were not retrieving the guest annotations before

**Comment by fxierh** [comment]
Could you also remove the get-guest-annotation step from the cucushift-installer-rehearse-azure-aks-hypershift-full-cert-guest workflow ? Thanks.

**Comment by Patryk-Stefanski** [comment]
> Could you also remove the get-guest-annotation step from the cucushift-installer-rehearse-azure-aks-hypershift-full-cert-guest workflow ? Thanks.

We don't use it there [cucushift-installer-rehearse-azure-aks-hypershift-full-cert-guest-workflow.yaml](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/installer/rehearse/azure/aks/hypershift/full-cert/guest/cucushift-installer-rehearse-azure-aks-hypershift-full-cert-guest-workflow.yaml)

**Comment by fxierh** [comment]
I mean the cucushift-installer-rehearse-azure-aks-hypershift-base-provision chain which in invoked as a part of the full-cert workflow

**Comment by fxierh** [comment]
Looks good. Let me launch a rehearse real quick

/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7

**Comment by fxierh** [comment]
The failures in the last rehearsal are unrelated to this change and shouldn't block the merge. Please feel free to unhold. 

/pj-rehearse ack
/hold

**Pull request merged**